### ISSUE TITLE
cmake support

### ISSUE THREAD

**Issue by nwalfield** [rpm-software-management/rpm-sequoia#37]
given that RPM is the main consumer for this, whether it wouldn't make sense to generate a CMake configuration (rpm-sequoia-config.cmake and rpm-sequoia-targets.cmake files) in the build output dir? CMake actually supports linking library targets with no SONAME (you just have to set the IMPORTED_NO_SONAME property TRUE on the IMPORTED SHARED target), so the symlink wouldn't even be necessary to make linking from the builddir work. You'd just do a cmake -DCMAKE_PREFIX_PATH=/path/to/rpm-sequoia/targets/<profile>/ when configuring CMake, which would use find_package(rpm-sequoia) instead of pkg_check_modules() to import it.

**Comment by nwalfield** [comment]
Adding cmake support would be a fine addition.  I'm not going to do it any time soon, as I don't know any cmake and I don't have time to learn it right now.  One thing to be aware of: although rpm has changed to cmake in their development branch, 4.18 still uses autoconf.

**Comment by pmatilai** [comment]
Yup, a more cmake-native configuration may well make sense ... once we stop caring about updates for 4.18.x. Right now there's not much point, and pkg-config is native enough for both worlds.
